packages/atoms/src/Switch.tsx

```diff
@@ -1,55 +1,82 @@
-import React from 'react';
-import { TouchableOpacity } from 'react-native';
+import React, { useEffect } from 'react';
+import Animated, {
+    Easing,
+    useAnimatedStyle,
+    useSharedValue,
+    interpolateColor,
+    withTiming,
+} from 'react-native-reanimated';
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Box } from './Box';
+import { Pressable } from 'react-native';
 
 type SwitchProps = {
     isChecked: boolean;
     onChange: (value: boolean) => void;
     isDisabled?: boolean; // Functionality of disabled works but styles are not implemented yet (waiting for design)
 };
 
-type SwitchStyleProps = {
-    isChecked: boolean;
-};
-const wrapperStyle = prepareNativeStyle<SwitchStyleProps>((utils, { isChecked }) => ({
-    width: 44,
-    height: 24,
-    justifyContent: 'center',
-    alignItems: 'flex-start',
+const SWITCH_CONTAINER_WIDTH = 44;
+const SWITCH_CONTAINER_HEIGHT = 24;
+const SWITCH_CIRCLE_SIZE = 20;
+const SWITCH_CIRCLE_MARGIN = 2;
+const TRACK_CIRCLE_WIDTH = SWITCH_CONTAINER_WIDTH - SWITCH_CIRCLE_SIZE - SWITCH_CIRCLE_MARGIN * 2;
+
+const switchContainerStyle = prepareNativeStyle(utils => ({
+    width: SWITCH_CONTAINER_WIDTH,
+    height: SWITCH_CONTAINER_HEIGHT,
     backgroundColor: utils.colors.gray400,
     borderColor: utils.colors.forest,
     borderRadius: utils.borders.radii.round,
-    extend: {
-        condition: isChecked,
-        style: {
-            alignItems: 'flex-end',
-            backgroundColor: utils.colors.green,
-        },
-    },
+    flexDirection: 'row',
 }));
 
-const SWITCH_CIRCLE_SIZE = 20;
-
 const switchCircleStyle = prepareNativeStyle(utils => ({
     width: SWITCH_CIRCLE_SIZE,
     height: SWITCH_CIRCLE_SIZE,
     backgroundColor: utils.colors.white,
     borderRadius: utils.borders.radii.round,
-    margin: 2,
+    margin: SWITCH_CIRCLE_MARGIN,
+    alignSelf: 'center',
 }));
 
 export const Switch = ({ isChecked, onChange, isDisabled = false }: SwitchProps) => {
-    const { applyStyle } = useNativeStyles();
+    const { applyStyle, utils } = useNativeStyles();
```

Comment:
what do you think about placing code from lines 43-63 to hook? I would suggest to create hook in this file called and then use it like `const {animatedSwitchCircleStyle, animatedSwitchContainerStyle} = useAnimationStyles({ isChecked })`

What do you think about this idea? My biggest concern is that in some more complicated components we will add lot of noise code to component just because of animation.

Reply:
Make sense to me: https://github.com/trezor/trezor-suite/pull/5546/commits/326652e453bb76b6d160db4ef5ffcd1a85d2c91f